website/static/js/profile.js

```diff
@@ -23,7 +23,13 @@ var socialRules = {
     github: /github\.com\/(\w+)/i,
     researchGate: /researchgate\.net\/profile\/(\w+)/i,
     academia: /(\w+)\.academia\.edu\/(\w+)/i,
-    baiduScholar: /xueshu\.baidu\.com\/scholarID\/(\w+)/i
+    baiduScholar: /xueshu\.baidu\.com\/scholarID\/(\w+)/i,
+    url: '^(https?:\\/\\/)?'+ // protocol
```

Comment:
Looks like it does. `from modularodm.validators import URLValidator` What we will want to avoid is to have someone change this from under us without knowing about it so we can update this, because having incompatible validators would be much worse than not validating the information. I think, in this case, we should ensure that our validation tests are reasonably complete as far as coverage and should be applied to both functions equally. 

So perhaps beef up your JS tests to handle query strings, ip addresses, ports, paths, protocols, etc and then make sure we have python tests that call URL validator that match those tests. The front end can be more restrictive than the back-end, if they have to be different.